bndtools.core/src/bndtools/central/Central.java

```diff
@@ -239,11 +238,21 @@ public synchronized static Workspace getWorkspace() throws Exception {
         }
     }
 
-    public synchronized static void onWorkspaceInit(Function<Workspace,Void> callback) {
-        if (workspace != null)
-            callback.run(workspace);
-        else
-            workspaceInitCallbackQueue.add(callback);
+    public static void onWorkspaceInit(Function<Workspace,Void> callback) {
+        callbacks.add(callback);
+        notifyCallbacks();
+    }
+
+    private static void notifyCallbacks() {
+        synchronized (instance) {
+            if (workspace == null)
+                return;
+        }
+
+        while (!callbacks.isEmpty()) {
```

Comment:
This loop is dangerous because you separate the isEmpty test from the poll call. This thread could see isEmpty is false, then another thread could remove the only item, then this thread could call poll, get null and then NPE at the apply call.

Safer is to do:

```
for (Function<Workspace,Void> callback = callbacks.poll(); callback != null; callback = callbacks.poll()) {
```